File: 2441/2441.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxK = function (nums) {
  let result = -1;
  let max = 0;

  for (const num of nums) {
    if (Math.abs(num) > max) {
      if (nums.indexOf(-num) > -1) {
        max = Math.abs(num);
        result = max;
      }
    }
  }
  return result;
};

File: 2441/2441.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int lo = 0;
        int hi = size(nums) - 1;
        while(lo<hi){
            if(-nums[lo] == nums[hi]){
                return nums[hi];
            }
            else if(-nums[lo]>nums[hi]){
                lo++;
            }
            else{
                hi--;
            }
        }
        return -1;
    }
};